CMakeLists.txt

```diff
@@ -0,0 +1,474 @@
+cmake_minimum_required (VERSION 2.8)
+project (logcabin)
+set (LOGCABIN_VERSION_MAJOR 1)
+set (LOGCABIN_VERSION_MINOR 2)
+
+include_directories(BEFORE .)
+include_directories(BEFORE ./include)
+include_directories(BEFORE ./build)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-strict-overflow -fPIC")
+
+# TODO: accept user input
+set(CMAKE_PROTO_CXX_FLAGS " ")
+set(CMAKE_GTEST_CXX_FLAGS " ")
+
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -DDEBUG")
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O2 -DNDEBUG")
+
+file(MAKE_DIRECTORY build)
+
+cmake_policy(SET CMP0015 NEW)
+link_directories(build)
+
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/build)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/build)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/build)
+
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    message("using clang")
+
+    ## clang will always use c++11 as std
+    set(CPP_STD c++11)
+
+    set (FLAG_LIST 
+        -Wno-c++98-compat-pedantic 
+        -Wno-covered-switch-default 
+        -Wno-deprecated 
+        -Wno-disabled-macro-expansion 
+        -Wno-documentation-unknown-command 
+        -Wno-exit-time-destructors 
+        -Wno-float-equal 
+        -Wno-global-constructors 
+        -Wno-gnu-zero-variadic-macro-arguments 
+        -Wno-missing-noreturn 
+        -Wno-missing-prototypes 
+        -Wno-missing-variable-declarations 
+        -Wno-packed 
+        -Wno-padded 
+        -Wno-reserved-id-macro 
+        -Wno-shadow 
+        -Wno-shift-sign-overflow 
+        -Wno-switch-enum 
+        -Wno-undef 
+        -Wno-unknown-warning-option 
+        -Wno-unused-macros 
+        -Wno-unused-member-function 
+        -Wno-unused-parameter 
+        -Wno-used-but-marked-unused 
+        -Wno-vla 
+        -Wno-vla-extension 
+        -Wno-weak-vtables
+        -Weverything
+        -Wno-unreachable-code ## clang 3.4 is knon to emit warnings without -Wno-unreachable-code, I will move this out when I have idea to get compiler version
+    )
+    foreach(FLAG in FLAG_LIST)
+        set(CMAKE_LOGCABIN_CXX_FLAGS  "${LOGCABIN_CXX_FLAGS} ${FLAG}" ) 
+    endforeach()
+elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    message("using gcc")
+
+    ## detect gcc version to decide wich c std should we use
```

Comment:
I am using 2.7 version of cmake and this var is not available on this version .on ubuntu12.04, this 2.7 is shipped by apt